service/resource/cloudformation/main_stack.go

```diff
@@ -2,19 +2,27 @@ package cloudformation
 
 import (
 	"bytes"
+	"strconv"
 	"text/template"
 
-	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/aws-operator/service/key"
 	"github.com/giantswarm/awstpr"
 )
 
 func newMainStack(customObject awstpr.CustomObject) (StackState, error) {
 	stackName := key.MainStackName(customObject)
+	workers := len(customObject.Spec.AWS.Workers)
+	var imageID string
+	if workers > 0 {
+		imageID = customObject.Spec.AWS.Workers[0].ImageID
```

Comment:
This looks like we have to adapt the TPR a bit. Lets make sure this is tracked and done.

Reply:
I've created https://github.com/giantswarm/awstpr/issues/47, added to sig-operator inbox.

Reply:
This is a long running issue. `clustertpr` has an entry for each worker node and the same approach was used in `awstpr`. Since we now manage workers as a pool with the ASG it causes this problem.

I'd say we should raise an issue and fix this as a separate task.